pennylane/devices/qubit/simulate.py

```diff
@@ -97,9 +114,8 @@ def simulate(circuit: qml.tape.QuantumScript, rng=None, debugger=None) -> Result
         for mp in circuit.measurements
     )
 
-    # no shot vector
-    if not circuit.shots.has_partitioned_shots:
-        return results
+    if circuit.shots.has_partitioned_shots:
+        # shot vector case: move the shot vector axis before the measurement axis
+        results = tuple(zip(*results))
 
-    # shot vector case: move the shot vector axis before the measurement axis
-    return tuple(zip(*results))
+    return (results, state, is_state_batched) if return_final_state else results
```

Comment:
Do we need to return `is_state_batched` given adjoint jacobian can't handle batched states?

Reply:
I'm going to revamp this based on the discussion with @timmysilv above. But to answer the question, I intended this function for general use and not just for adjoint, and other callers may want this information.